main.tf

```diff
@@ -45,6 +44,23 @@ resource "aws_subnet" "private" {
   tags              = "${merge(var.tags, map("Name", format("%s-subnet-private-%s", var.name, element(var.azs, count.index))))}"
 }
 
+resource "aws_subnet" "database" {
+  vpc_id            = "${aws_vpc.mod.id}"
+  cidr_block        = "${var.database_subnets[count.index]}"
+  availability_zone = "${element(var.azs, count.index)}"
+  count             = "${length(var.database_subnets)}"
+  tags              = "${merge(var.tags, map("Name", format("%s-database-subnet-%s", var.name, element(var.azs, count.index))))}"
+}
+
+resource "aws_db_subnet_group" "database" {
+  name        = "rds-subnet-group"
```

Comment:
include `${var.name}` into `name` to be able to have multiple `rds-subnet-group` without name collision.